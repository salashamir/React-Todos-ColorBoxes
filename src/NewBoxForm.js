import { useState } from "react";
import { v4 as uuid } from "uuid";

export const NewBoxForm = ({ addNewBox }) => {
  const INITIAL_STATE = { height: "", width: "", backgroundColor: "" };
  const [formData, setFormData] = useState(INITIAL_STATE);

  const handleChange = (evt) => {
    const { name, value } = evt.target;
    setFormData((formData) => {
      return { ...formData, [name]: value };
    });
  };

  const handleSubmit = (evt) => {
    evt.preventDefault();
    addNewBox({ ...formData, id: uuid() });
    setFormData(INITIAL_STATE);
  };
  return (
    <div>
      <form onSubmit={handleSubmit}>
        <div>
          <label htmlFor="width">Width:</label>
          <input
            onChange={handleChange}
            type="number"
            name="width"
            id="width"
            value={formData.width}
          />
        </div>
        <div>
          <label htmlFor="height">Height:</label>
          <input
            onChange={handleChange}
            type="number"
            name="height"
            id="height"
            value={formData.height}
          />
        </div>
        <div>
          <label htmlFor="backgroundColor">Color:</label>
          <input
            onChange={handleChange}
            type="text"
            name="backgroundColor"
            id="backgroundColor"
            value={formData.backgroundColor}
          />
        </div>
        <button type="submit">Add new</button>
      </form>
    </div>
  );
};

export default NewBoxForm;
