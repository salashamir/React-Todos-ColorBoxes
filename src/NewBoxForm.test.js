import { render } from "@testing-library/react";
import NewBoxForm from "./NewBoxForm";

it("should render without crashing", () => {
  render(<NewBoxForm />);
});

it("should match snapshot", () => {
  const { asFragment } = render(<NewBoxForm />);
  expect(asFragment()).toMatchSnapshot();
});
