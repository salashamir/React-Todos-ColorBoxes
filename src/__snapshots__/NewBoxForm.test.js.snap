// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div>
    <form>
      <div>
        <label
          for="width"
        >
          Width:
        </label>
        <input
          id="width"
          name="width"
          type="number"
          value=""
        />
      </div>
      <div>
        <label
          for="height"
        >
          Height:
        </label>
        <input
          id="height"
          name="height"
          type="number"
          value=""
        />
      </div>
      <div>
        <label
          for="backgroundColor"
        >
          Color:
        </label>
        <input
          id="backgroundColor"
          name="backgroundColor"
          type="text"
          value=""
        />
      </div>
      <button
        type="submit"
      >
        Add new
      </button>
    </form>
  </div>
</DocumentFragment>
`;
